src/NuGet.Core/NuGet.Build.Tasks/GetRestoreNuGetAuditSuppressionsTask.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Newtonsoft.Json;
+
+namespace NuGet.Build.Tasks
+{
+    public class GetRestoreNuGetAuditSuppressionsTask : Microsoft.Build.Utilities.Task
+    {
+        /// <summary>
+        /// Full path to the msbuild project.
+        /// </summary>
+        [Required]
+        public string ProjectUniqueName { get; set; }
+
+        [Required]
+        public ITaskItem[] NuGetAuditSuppressions { get; set; }
+
+        /// <summary>
+        /// Target frameworks to apply this for. If empty this applies to all.
+        /// </summary>
+        public string TargetFrameworks { get; set; }
+
+        /// <summary>
+        /// Output items
+        /// </summary>
+        [Output]
+        public ITaskItem[] RestoreGraphItems { get; set; }
+
+        public override bool Execute()
+        {
+            if (NuGetAuditSuppressions.Length == 0) return true;
+
+            var entries = new List<ITaskItem>(NuGetAuditSuppressions.Length);
+            var seenIds = new HashSet<string>(NuGetAuditSuppressions.Length, StringComparer.OrdinalIgnoreCase);
```

Comment:
```suggestion
            var seenIds = new HashSet<string>(NuGetAuditSuppressions.Length, StringComparer.Ordinal);
```

I don't think it's a safe assumption that URLs are always case insensitive. While GitHub Advisories Database is, a generic web server can serve static files from a filesystem. And some filesystems are case sensitive. Therefore, if the filesystem contains both `index.html` and `Index.html`, which file should be served if the request comes in for `INDEX.HTML`?

Reply:
That's fair. I've made this change now, and added a test for it as well